### PR TITLE
libSplash: no HDF5 Fortran

### PR DESCRIPTION
**Pull request by ax3l**
libSplash is a C++-only library that does not depend on the Fortran bindings of HDF5.

Also reduces potential points that can (and often do) break in the dependencies.

### CODE REVIEW

**Comment by ax3l** [comment]
@davydden yes, that is true. I was willing to take that double-variant into account since HDF5 Fortran breaks too often with MPI.

As a work-around, I tried to write
```python
depends_on('libsplash@1.6.0 ^hdf5~fortran', when='+hdf5')
```
in a downstream dependency, but for some reason it still tried to build it with Fortran.

**Comment by davydden** [comment]
>  I was willing to take that double-variant in a user's install into account since HDF5 Fortran breaks too often with MPI

maybe use site or user configuration files to do what you want http://spack.readthedocs.io/en/latest/configuration.html?highlight=configuration#configuration-scopes ?

**Comment by ax3l** [comment]
Yes, changing the defaults of the `Concretization Preferences` would do.

For this C++-only package, that has and depends on no Fortran capabilities, building all the HDF5 & MPI Fortran bindings by default is still a bit odd.

Seems I can not express "don't add that dependencies variant by default but don't care if they are on" in the libSplash package.py right now?

**Comment by davydden** [comment]
>  building all the HDF5 & MPI Fortran bindings by default is still a bit odd.

i think you misinterpret what the line says. In you current PR you say: only HDF5 **without** fortran can be used with this package. 

Whether or not `hdf5` by defaults build with `fortran`, is up to you settings and hdf5 package settings. Is is not up to `libSplash`. 

Think of it in a different way: say I use `libSplash` with other libs which need `hdf5+fortran`. You changes will make it impossible, which is wrong.

Bottom line: I think you are fixing the problem in the wrong place.

**Comment by ax3l** [comment]
Yep, I see.
Thanks for the help! :)

**Comment by ax3l** [comment]
@davydden btw, any idea why a downstream package's
```python
depends_on('libsplash@1.6.0 ^hdf5~fortran', when='+hdf5')
```

doesn't work? syntax error?

**Comment by davydden** [comment]
Probably, I never saw depends_on being used this way.
